resources/js/components/fieldtypes/grid/Grid.vue

```diff
@@ -207,6 +211,30 @@ export default {
         removed(index) {
             if (! confirm(__('Are you sure?'))) return;
 
+            let errors = this.storeState.errors || {};
+
+            errors = Object.keys(errors).reduce((acc, key) => {
+                if (key.startsWith(`${this.fieldPathPrefix || this.handle}.${index}.`)) {
+                    return acc;
+                }
+
+                if (key.startsWith(`${this.fieldPathPrefix || this.handle}.`)) {
+                    const parts = key.split('.');
+                    const rowIndex = parseInt(parts[1], 10);
+
+                    if (rowIndex > index) {
+                        parts[1] = (rowIndex - 1).toString();
+                        acc[parts.join('.')] = errors[key];
+                    } else {
+                        acc[key] = errors[key];
+                    }
+                }
+
+                return acc;
+            }, {});
+
+            this.$store.commit(`publish/${this.storeName}/setErrors`, errors);
```

Comment:
It looks like `publish/storeName/setErrors` updates the errors in the Vuex state, but it doesn't affect the `errors` property in the `PublishForm` component.... not sure if that'll end up being an issue? 🤔